presto-main/src/main/java/com/facebook/presto/operator/scalar/annotations/ParametricScalarImplementation.java

```diff
@@ -313,12 +328,24 @@ private void parseArguments(Method method)
                     Annotation[] annotations = parameter.getAnnotations();
                     checkArgument(!Stream.of(annotations).anyMatch(IsNull.class::isInstance), "Method [%s] has @IsNull parameter that does not follow a @SqlType parameter", method);
 
-                    SqlType type = Stream.of(annotations)
-                            .filter(SqlType.class::isInstance)
-                            .map(SqlType.class::cast)
-                            .findFirst()
-                            .orElseThrow(() -> new IllegalArgumentException(format("Method [%s] is missing @SqlType annotation for parameter", method)));
-                    TypeSignature typeSignature = parseTypeSignature(type.value(), literalParameters);
+                    if (Stream.of(annotations).anyMatch(BlockPosition.class::isInstance) || Stream.of(annotations).anyMatch(BlockIndex.class::isInstance)) {
+                        BlockPosition typeBP = Stream.of(annotations)
```

Comment:
Is this code change directly related to moving `BlockPosition` and `BlockIndex` annotations to SPI?